controllers/kyma_controller.go

```diff
@@ -286,15 +289,13 @@ func (r *KymaReconciler) HandleProcessingState(ctx context.Context, kyma *v1alph
 	return nil
 }
 
-func (r *KymaReconciler) HandleDeletingState(ctx context.Context, kyma *v1alpha1.Kyma) (bool, error) {
+func (r *KymaReconciler) HandleDeletingState(ctx context.Context, kyma *v1alpha1.Kyma,
+	syncContext *remote.KymaSynchronizationContext,
+) (bool, error) {
 	logger := log.FromContext(ctx)
 
-	if kyma.Spec.Sync.Enabled {
-		syncContext, err := remote.InitializeKymaSynchronizationContext(ctx, kyma, r.Client, r.RemoteClientCache)
-		if err != nil {
-			return false, fmt.Errorf("remote sync initialization failed: %w", err)
-		}
-		if err = r.RemoveWebhookChart(ctx, kyma, syncContext); err != nil {
+	if kyma.Spec.Sync.Enabled && syncContext != nil {
+		if err := r.RemoveWebhookChart(ctx, syncContext); err != nil {
```

Comment:
why has remove web hook chart no dedication switch between recoverable and non-recoverable errors? e.g. a chart installation is not something that can necessarily be recovered simply by waiting (and returning true, nil as you do)

Reply:
agree on this, no way for us here.